analytics/lib/counts.py

```diff
@@ -343,16 +347,26 @@ def do_increment_logging_stat(
     else:
         raise AssertionError("Unsupported CountStat frequency")
 
-    row, created = table._default_manager.get_or_create(
-        property=stat.property,
-        subgroup=subgroup,
-        end_time=end_time,
-        defaults={"value": increment},
-        **id_args,
+    id_args_keys = SQL(", ").join(map(Identifier, id_args.keys()))
+    values = SQL(", ").join(map(Literal, id_args.values()))
+    sql_query = SQL(
+        """
+        INSERT INTO {table_name}(property, subgroup, end_time, value, {id_args_keys})
+        VALUES (%s, %s, %s, %s, {id_args_values})
+        ON CONFLICT (property, subgroup, end_time, {id_args_keys_update})
```

Comment:
I think the issue is that we don't have one index -- we have two partial indexes, on subgroup being null or not-null.  Per the `index_predicate` part of https://www.postgresql.org/docs/current/sql-insert.html I think that means that this line needs to either be:
```
        ON CONFLICT (property, subgroup, end_time, {id_args_keys_update}) WHERE subgroup IS NOT NULL
```
...or:
```
        ON CONFLICT (property, end_time, {id_args_keys_update}) WHERE subgroup IS NULL
```
...based on if the `subgroup` Python variable is None or not.

It's possible that the latter can be:
```
        ON CONFLICT (property, subgroup, end_time, {id_args_keys_update}) WHERE subgroup IS NULL
```
...but you'd need to test.

Reply:
```
> ...based on if the subgroup Python variable is None or not.
```
could we use simple if-else block to dynamically generate SQL for WHERE clause